smart-metastore/src/main/java/org/smartdata/metastore/dao/SystemInfoDao.java

```diff
@@ -0,0 +1,128 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.smartdata.metastore.dao;
+
+import org.apache.commons.lang.StringUtils;
+import org.smartdata.model.CmdletInfo;
+import org.smartdata.model.FileDiff;
+import org.smartdata.model.FileDiffState;
+import org.smartdata.model.FileDiffType;
+import org.smartdata.model.FileInfo;
+import org.smartdata.metastore.utils.MetaStoreUtils;
+import org.smartdata.model.SystemInfo;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+
+import javax.sql.DataSource;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SystemInfoDao {
+  private final String TABLE_NAME = "sys_info";
+  private DataSource dataSource;
+
+  public void setDataSource(DataSource dataSource) {
+    this.dataSource = dataSource;
+  }
+
+  public SystemInfoDao(DataSource dataSource) {
+    this.dataSource = dataSource;
+  }
+
+  public List<SystemInfo> getAll() {
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+    return jdbcTemplate.query("select * from " + TABLE_NAME, new SystemInfoRowMapper());
+  }
+
+  public List<SystemInfo> getByProperty(String property) {
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+    return jdbcTemplate.query("select * from " + TABLE_NAME + " where property = ?",
+        new Object[]{property}, new SystemInfoRowMapper());
+  }
+
+  public List<SystemInfo> getByProperties(List<String> properties) {
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+    return jdbcTemplate.query("select * from " + TABLE_NAME + " where property IN (?)",
+        new Object[]{StringUtils.join(properties, ",")},
+        new SystemInfoRowMapper());
+  }
+
+  public void delete(String property) {
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+    final String sql = "delete from " + TABLE_NAME + " where property = ?";
+    jdbcTemplate.update(sql, property);
+  }
+
+  public void insert(SystemInfo systemInfo) {
```

Comment:
When inserting a SystemInfo, we should first check whether the key already exists, if so, throw an Exception.